cpp/examples/build.sh

```diff
@@ -1,14 +1,15 @@
 #!/bin/bash
 
-# Copyright (c) 2021, NVIDIA CORPORATION.
+# Copyright (c) 2021-2022, NVIDIA CORPORATION.
 
 # libcudf examples build script
 
 # Parallelism control
 PARALLEL_LEVEL=${PARALLEL_LEVEL:-4}
 
 # Root of examples
-EXAMPLES_DIR=${WORKSPACE}/cpp/examples
+EXAMPLES_DIR=$(dirname "$(realpath "$0")")
+LIB_BUILD_DIR=${LIB_BUILD_DIR:-$(readlink -f "${EXAMPLES_DIR}/../build")}
```

Comment:
When do we need to support examples built outside this tree? i.e. why do we need to check for a preexisting `LIB_BUILD_DIR` variable before applying the default?

Reply:
These changes were made to fix some build errors that occurred as a result of the recipe changes.

-  This first change was made to eliminate a dependency on the `WORKSPACE` environment variable, which is only ever set in CI. In CI, this change is a no-op, since `EXAMPLES_DIR` will evaluate to the same value. I don't think this ever worked locally unless users were explicitly setting the `WORKSPACE` variable (doubtful), but at the very least this means that the `WORKSPACE` variable won't have to be set anymore for this script to work correctly.
  ```diff
- EXAMPLES_DIR=${WORKSPACE}/cpp/examples
+ EXAMPLES_DIR=$(dirname "$(realpath "$0")")
  ```

- The `LIB_BUILD_DIR` value was originally introduced [here](https://github.com/rapidsai/cudf/pull/8638/files#diff-a572e8c9004b6b341c6724ca772791ddee762c47ed32022863bbd02bee6cdab3R21), but I don't think it worked as intended since the value was not actually being set anywhere in CI. So the change I proposed here maintains the default `LIB_BUILD_DIR` value if it's set, but if it's not set will revert to a sensible default. Without this change, the new conda recipe build was breaking because it couldn't find the locally built `cudf` root directory.
```diff
+ LIB_BUILD_DIR=${LIB_BUILD_DIR:-$(readlink -f "${EXAMPLES_DIR}/../build")}
```